healthcare.py

```diff
@@ -0,0 +1,1949 @@
+-----------
+Context:
+According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
+This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, 
+and smoking status. Each row in the data provides relavant information about the patient.
+
+Attribute Information
+1) id: unique identifier
+2) gender: "Male", "Female" or "Other"
+3) age: age of the patient
+4) hypertension: 0 if the patient doesn't have hypertension, 1 if the patient has hypertension
+5) heart_disease: 0 if the patient doesn't have any heart diseases, 1 if the patient has a heart disease
+6) ever_married: "No" or "Yes"
+7) work_type: "children", "Govt_jov", "Never_worked", "Private" or "Self-employed"
+8) Residence_type: "Rural" or "Urban"
+9) avg_glucose_level: average glucose level in blood
+10) bmi: body mass index
+11) smoking_status: "formerly smoked", "never smoked", "smokes" or "Unknown"*
+12) stroke: 1 if the patient had a stroke or 0 if not
+*Note: "Unknown" in smoking_status means that the information is unavailable for this patient
+
+
+About the model: 
+-The model does not contain data for ages < 18 and BMI > 60 f
+
+'''
+#########################################################################################################################################################
+
+'''
+Section 2.0
+-----------
+Import all required libraries 
+'''
+import pandas as pd 
+import numpy as np 
+import matplotlib.pyplot as plt 
+import seaborn as sns
+
+#########################################################################################################################################################
+'''
+Section 3.0
+-----------
+Import the data
+'''
+df = pd.read_csv("C:\\Users\\PZ4L6Q\\Documents\\Learnings\\Projects\\Project_1\\healthcare-dataset-stroke-data.csv")
+# print(df.head())
+
+#########################################################################################################################################################
+'''
+Section 4.0
+-----------
+Perform (Exploratory Data Analysis) EDA on the data. 
+'''
+
+
+'''
+Section 4.1
+-----------
+Drop columns if required
+'''
+df = df.drop('id', axis =1 ) #-> this column will not be useful during the training of the ML algorithm 
+# print(df.head())
+
+
+'''
+Section 4.2
+-----------
+Analyze Each column & View the unique columns of 'SELECTED' columns . 
+    Then, go over each column and to replace all of the white spaces by NaN.
+    Finally, make a conclusion in regards of each of the 'SELECTED analyzed the columns
+    Selected columns ->  gender, hypertension, heart_disease, ever_married, work_type, Residence_type, avg_glucose_level, bmi, smoking_status, stroke
+    Unselected colummns (requires individual analyzation ) -> age, avg_glucose_level
+
+'''
+
+# df['gender'] = df['gender'].replace(r'^\s*$', np.nan, regex=True)

```

Comment:
What is this doing?

Reply:
replacing the blank values with nan 
ref: https://stackoverflow.com/questions/13445241/replacing-blank-values-white-space-with-nan-in-pandas